skymp5-server/cpp/server_guest_lib/PieScript.cpp

```diff
@@ -432,10 +446,30 @@ void PieScript::Play(MpActor& actor, const WorldState& worldState,
                      FormDesc::FromFormId(wardrobeId, worldState.espmFiles) });
   }
 
-  auto it = miscLootTable.find(itemBaseId);
-  if (it != miscLootTable.end()) {
+  if (auto it = miscLootTable.find(itemBaseId); it != miscLootTable.end()) {
     for (const auto& item : miscLootTable[itemBaseId]) {
       AddItem(actor, worldState, item, 1);
     }
   }
+
+  if (auto it = bookBoundWeapons.find(itemBaseId);
+      it != bookBoundWeapons.end()) {
+    float currentMagickaPercentage =
+      actor.GetChangeForm().actorValues.magickaPercentage;
+    if (currentMagickaPercentage >= it->second.GetManacostPercentage()) {
+      actor.DamageActorValue(espm::ActorValue::Magicka,
+                             it->second.GetManacost());
+      uint32_t boundWeaponBaseId = it->second.GetBaseId(),
+               bookBaseId = it->first;
+      actor.AddItem(boundWeaponBaseId, 1);
+      actor.RemoveItem(bookBaseId, 1, nullptr);
+      worldState.SetTimer(it->second.GetCooldown())
+        .Then([bookBaseId, boundWeaponBaseId, &actor](Viet::Void) {
```

Comment:
&actor can be destroyed at this point. so instead of & we should use formId and =

Reply:
+ added items should be also equipped